lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,323 @@
+            data_disk_size=search_space.IntRange(min=1),
+        )
+        node_capabilities.network_interface = schema.NetworkInterfaceOptionSettings()
+        node_capabilities.network_interface.max_nic_count = 1
+        node_capabilities.network_interface.nic_count = 1
+        node_capabilities.gpu_count = 0
+        node_capabilities.features = search_space.SetSpace[schema.FeatureSettings](
+            is_allow_set=True,
+            items=[
+                schema.FeatureSettings.create(SerialConsole.name()),
+            ],
+        )
+
+        return node_capabilities
+
+    def _deploy_environment(self, environment: Environment, log: Logger) -> None:
+        self._deploy_nodes(environment, log)
+
+    def _prepare_common_vhd(self, vhd_local_path: PurePath, log: Logger) -> None:
+        if self._common_vhd_path:
+            return
+
+        vhd_remote_path = PureWindowsPath(
+            self.server_node.working_path / f"common_vhd.{vhd_local_path.suffix}"
+        )
+
+        is_zipped = False
+        if vhd_local_path.suffix == ".zip":
+            is_zipped = True
+            vhd_remote_path = PureWindowsPath(
+                self.server_node.working_path / "zipped_vhd.zip"
+            )
+
+        log.debug("Copying VHD to server")
+        self.server_node.shell.copy(vhd_local_path, vhd_remote_path)
+        log.debug("Finished copying VHD to server")
+
+        if is_zipped:
+            vhd_remote_path = self._unzip_vhd(vhd_remote_path)
+
+        self._common_vhd_path = vhd_remote_path
+
+    def _deploy_nodes(self, environment: Environment, log: Logger) -> None:
+        if environment.runbook.nodes_requirement is None:
+            return  # nothing to deploy?
+
+        test_suffix = "".join(random.choice(string.ascii_uppercase) for _ in range(5))
+        vm_name_prefix = f"lisa-{test_suffix}"
+
+        hv = self.server_node.tools[HyperV]
+        default_switch = hv.get_first_switch()
+
+        extra_args = {
+            x.command.lower(): x.args for x in self._hyperv_runbook.extra_args
+        }
+
+        self.console_logger = SerialConsoleLogger(self.server_node)
+
+        for i, node_space in enumerate(environment.runbook.nodes_requirement):
+            node_runbook = node_space.get_extended_runbook(
+                HypervNodeSchema, type(self).type_name()
+            )
+
+            vm_name = f"{vm_name_prefix}-{i}"
+
+            node = environment.create_node_from_requirement(node_space)
+            assert isinstance(node, RemoteNode)
+
+            self._prepare_common_vhd(PurePath(node_runbook.vhd), log)
+            assert self._common_vhd_path
+
+            node.name = vm_name
+
+            node_context = get_node_context(node)
+            node_context.vm_name = vm_name
+            node_context.host = self.server_node
+
+            node_context.working_dir = PureWindowsPath(
+                self.server_node.working_path / f"{vm_name}"
```

Comment:
vm_name may be conflict, if it's specified in runbook. So the path should include test run part.

Reply:
There is no name field in the Hyper-V runbook schema.

Reply:
Ok. We will add name in runbook later. It's used to test with the existing VM.

Reply:
Run ID is already part of the `working_path`. What more do we need?

Reply:
Make sure environment is a part of the `working_path` too.

Reply:
environment name is part of vm name. `lisa-{environment.name}-n{index}`. This is like Azure platform: since we don't have resource group, I am using environment name.